### DISCUSSION TITLE
Sign Transaction and send raw transaction immediately

### DISCUSSION THREAD

**Discussion by dennisloh95** [MetaMask/snaps#2502] [Q&A]
Hi, is metamask or snaps can achieve user sign a transaction and send raw transaction? 

The context of this is, our protocol have a gas sponsor on protocol level, it helps users onboarding our protocol without native token, I've research few places and manage to get it done with eth_sign, however it is not recommend user to enable this deprecated method. So what I need to achieve is, user sign the transaction, but use rpc to send the raw transaction so that transaction is not broadcast by the user and pay the gas. It did not required any changes after sign.

Any way to achieve this?

**Comment by Montoya** [comment]
From a Snap you can do `personal_sign` and `eth_sendRawTransaction`, but you cannot do `eth_signTypedDatav4` which is probably what you need. Can you tell me which signing method you want to use?

**Reply by dennisloh95** [reply]
For example, after our backend create the transaction, and return to user,
we have
```
{
data: 0x00000000000000000000000000004D616E746C654D6574615478507265666978
gas: 567812796n
maxFeePerGas: 100000000n
nonce: 2
type: "eip1559"
}
```

the data pass from backend will have a prefix to prove that this transaction data is a gas sponsor transaction, what the last step is user need to sign this transaction, then next immediately let an ethClient to sendRawTransaction. I need a normal sign transaction that can broadcast immediately as rawTransaction. Possible to achieve this with personal_sign?

or possible to have a signAndSendRawTransaction method?

**Reply by Balmasexy** [reply]
Good
